concepts/definitions.py

```diff
@@ -898,3 +898,20 @@ def shape(self) -> _common.Shape:
             Shape(objects=2, properties=1)
         """
         return _common.Shape._from_pair(self.objects, self.properties)
+    
+    @property
+    def density(self) -> float:
+        """The density of the definition.
+
+        Returns:
+            Definition density.
+        
+        Example:
+            >>> import concepts
+            >>> definition = concepts.Definition(['King Arthur', 'grail'],
+            ...                                  ['holy', 'historic'],
+            ...                                  [(False, True), (True, True)])
+            >>> definition.density
+            0.75
+        """
+        return sum(map(sum, self.bools)) / (self.shape.rows * self.shape.columns)
```

Comment:
> sum(map(sum, self.bools))

Sounds expensive:  isn't this the same as `len(self._pairs)`?

> (self.shape.rows * self.shape.columns)

Calculates shape twice (regular non-cached property).